cli/src/commands/status.rs

```diff
@@ -212,3 +218,60 @@ pub(crate) fn cmd_status(
 
     Ok(())
 }
+
+fn print_collapsed_untracked_files(
+    formatter: &mut dyn Formatter,
+    untracked_paths: &BTreeMap<RepoPathBuf, UntrackedReason>,
+    tree: &MergedTree,
+    path_converter: &RepoPathUiConverter,
+) -> io::Result<()> {
+    formatter.with_label("diff", |formatter| {
+        let tracked = tree
+            .entries()
+            .map(|entry| entry.0.into_internal_string())
+            .collect::<BTreeSet<String>>();
+
+        // TODO: This loop can be improved with BTreeSet cursors once that's stable,
+        // would remove the need for the whole `skip_prefixed_by` thing and turn it
+        // into a BTree lookup.
+        let mut skip_prefixed_by_dir = None;
+        for original_path in untracked_paths.keys() {
+            if skip_prefixed_by_dir
+                .as_ref()
+                .is_some_and(|p| original_path.as_internal_file_string().starts_with(p))
```

Comment:
It's probably better to compare paths as `&RepoPath`, not as `&str`.

Reply:
Can't really be done because `RepoPath` can't represent paths with a trailing `/`. This comparison only works because `skip_prefixed_by_dir` contains a trailing `/`, otherwise it would cause false positives if there is a tree like this:
```
directory/
directory_2/file
```
then printing `directory_2/file` would be skipped because it starts with `directory`, this doesn't happen if we check whether it starts with `directory/`.
At least I think this would happen unless I'm missing something.

Reply:
I'm not sure what the underlying problem is, but `RepoPath("directory_2/file").starts_with(RepoPath("directory"))` should be false (because they don't share the common prefix.)

Reply:
Oh you're right I didn't notice it does actually check for the `/`. That should be a bit cleaner then, thanks.